### DISCUSSION TITLE
Wallpaper above windows

### DISCUSSION THREAD

**Discussion by ItsLiyua** [hyprwm/Hyprland#1432] [Q&A]
Hey yall.
I've just installed Hyprland and tried setting a wallpaper. I'm using a wallpaper utility called [Linux WallpaperEngine](https://github.com/Almamu/linux-wallpaperengine). I added an exec-once line in the `hyprland.conf`-file that runs it (`exec-once = linux-wallpaperengine 1623047205`) It starts fine however when I launch an app like kitty it's being displayed below my wallpaper. I know because the windows shines through a little. When I use the close hotkey it closes the window first. When I use it again it kills the wallpaper utility and leaves me with the gray default background.

I tried using both `exec` and `exec-once` with the command both in front of all the other `exec` statements and after all of them. Neither showed any difference. 

Something I'd like to add: waybar is being displayed above the wallpaper and I don't know how. 

If you have any idea please tell me. Thanks for reading.

**Comment by vaxerski** [comment]
that utility is made for x11 and not wayland. It will _not_ work properly.

**Reply by ItsLiyua** [reply]
Since it runs like a window is there a way to have wayland put it behind applications? I mean it runs just fine. It's just not where I want it to be and telling windows where they should be is kind of the job of wayland/hyprland.

**Reply by vaxerski** [reply]
#1343

it's just _not_ the way we do it in wayland.
